lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -64,9 +76,21 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(
     }
     fUsdDstPath = parent->prim().GetPath().AppendChild(TfToken(childName));
 
-    fLayer = MayaUsdUtils::defPrimSpecLayer(childPrim);
-    if (!fLayer) {
-        std::string err = TfStringPrintf("No prim found at %s", childPrim.GetPath().GetString().c_str());
+    fChildLayer = MayaUsdUtils::defPrimSpecLayer(childPrim);
```

Comment:
Problems when parenting from prims defined in one layer to prims defined in another were because SdfCopySpec was being called with only one layer, the child layer.  Fixed to use both the child and parent layer.